app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt

```diff
@@ -237,6 +242,15 @@ class CtaViewModel @Inject constructor(
         }
     }
 
+    @WorkerThread
+    private fun getCtaOnSerp(): DaxDialogCta? {
+        return when {
+            !daxDialogSerpShown() -> DaxDialogCta.DaxSerpCta(onboardingStore, appInstallStore)
+            canShowWidgetDaxCta() -> DaxDialogCta.SearchWidgetCta(widgetCapabilities, onboardingStore, appInstallStore)
+            else -> null
```

Comment:
can we return here something like `DaxDialogCta.NotNeeded` instead of `null`?

Reply:
I agree with a different strategy to avoid returning `null` but I think we should agree first on a guideline or a tech design (using a new type inside `Cta` class like you suggested, and maybe `Option` in other cases).

Also, I think the change should be introduced in a different PR (LHF or inside the next project that comes here to introduce new Ctas). why? Because the change will affect how Ctas are handled in `BrowserTabViewModel`, `BrowserTabFragment`, and other methods in this same file + their own tests. The PR will be big enough to deserve its own PR so the reviewer can focus on the refactor.

what do you think?

Reply:
Nullability is a well-defined concept in kotlin, the code is understandable, not lying and not duplicating state. This is a good suggestion however I place it in the "nice to have" rather than PR blocking category.